src/stavebarline.ts

```diff
@@ -3,25 +3,35 @@
 // Author Larry Kuhns 2011
 
 import { Flow } from './tables';
-import { StaveModifier } from './stavemodifier';
-
+import { LayoutMetrics, StaveModifier } from './stavemodifier';
+import { Stave } from './stave';
+
+export enum BarlineType {
+  SINGLE = 1,
+  DOUBLE = 2,
+  END = 3,
+  REPEAT_BEGIN = 4,
+  REPEAT_END = 5,
+  REPEAT_BOTH = 6,
+  NONE = 7,
+}
 export class Barline extends StaveModifier {
-  static get CATEGORY() {
+  protected widths: Record<string, number>;
+  protected paddings: Record<string, number>;
+  protected layoutMetricsMap: Record<number, LayoutMetrics>;
+
+  protected thickness: number;
+  protected type!: BarlineType;
+
+  static get CATEGORY(): string {
     return 'barlines';
   }
-  static get type() {
-    return {
-      SINGLE: 1,
-      DOUBLE: 2,
-      END: 3,
-      REPEAT_BEGIN: 4,
-      REPEAT_END: 5,
-      REPEAT_BOTH: 6,
-      NONE: 7,
-    };
+
+  static get type(): typeof BarlineType {
+    return BarlineType;
   }
 
-  static get typeString() {
+  static get typeString(): Record<string, number> {
```

Comment:
can we narrow the Record with literals so it's `Record<BarlineTypeString, BarlineType>`?

Reply:
How would you declare `BarlineTypeString` if we make `enum` we change the interface. 
`BarlineType` change done.

Reply:
@0xfe any suggestion on how to declare `BarlineTypeString`? If `string` is ok, this is ready to merge! :)